### ISSUE TITLE
After bundle update, carrier wave stopped uploading images from remote url api 

### ISSUE THREAD

**Issue by farooqch11** [carrierwaveuploader/carrierwave#2411]
NoMethodError: undefined method `[]' for nil:NilClass

**Comment by mshibuya** [comment]
Full stack trace please.

**Comment by farooqch11** [comment]
i downgraded to last version to make it work in production! but this is was not going.

**Comment by mshibuya** [comment]
I need details to track down the issue, what did you do and what happened? How does your code look like and how was the whole stack trace of the error?

**Comment by farooqch11** [comment]
i am using carrier wave with the API , once i use remote_image_url, it throws weird error with latest version! 
- update your bundle
-use api to send remote_url 
-it will break instead of downloading image and uploading again!

**Comment by mshibuya** [comment]
I'm telling you that I need **full stack trace**, please.
That's what you can see like this upon the error, by pressing `Full Trace`.
![](http://cdn-ak.f.st-hatena.com/images/fotolife/y/y_yagi_2/20140810/20140810074654.jpg)

**Comment by farooqch11** [comment]
@mshibuya  i dont have stack trace now, because it was happening on heroku and production! i had to update it quickly without having complaints!

 but i mentioned how to **regenerate** it!

**Comment by fukayatsu** [comment]
In my case, 

    undefined method `[]' for nil:NilClass

occurred at

https://github.com/carrierwaveuploader/carrierwave/blob/v2.0.1/lib/carrierwave/downloader/remote_file.rb#L28 

when the remote image response header includes `content-disposition: inline;filename=""` (from google auth image)

---

```ruby
[1] pry(main)> match = 'content-disposition: inline;filename=""'.match(/filename=(?:"([^"]+)"|([^";]+))/)
=> nil
[2] pry(main)> match[1].presence || match[2].presence
NoMethodError: undefined method `[]' for nil:NilClass
```


----

trace:

```
vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/downloader/remote_file.rb:28:in `filename_from_header': undefined method `[]' for nil:NilClass (NoMethodError)
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/downloader/remote_file.rb:11:in `original_filename'
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/sanitized_file.rb:43:in `original_filename'
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/sanitized_file.rb:57:in `filename'
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/uploader/cache.rb:133:in `cache!'
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/uploader/download.rb:20:in `download!'
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/mounter.rb:98:in `block (2 levels) in remote_urls='
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/mounter.rb:176:in `handle_error'
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/mounter.rb:96:in `block in remote_urls='
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/mounter.rb:95:in `each'
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/mounter.rb:95:in `remote_urls='
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/mount.rb:166:in `remote_icon_url='
    from vendor/bundle/ruby/2.6.0/gems/carrierwave-2.0.1/lib/carrierwave/orm/activerecord.rb:21:in `remote_icon_url='
...
```

**Comment by mshibuya** [comment]
Fixed by #2419.
